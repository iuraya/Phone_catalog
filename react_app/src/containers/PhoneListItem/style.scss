.phone-list-item {
  width: 100%;
  height: 100%;
  display: flex;
  margin-top: 10px;
  border-radius: 4px;
  border: 2px solid #41ADDD;
  font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
  // font-weight: bold;
  font-size: 16px;
  color: #41ADDD;


  img {
    height: 100px;
  }

}
