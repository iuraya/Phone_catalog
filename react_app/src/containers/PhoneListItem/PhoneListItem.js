/* eslint react/require-default-props: "warn" */
/* eslint react/forbid-prop-types: "warn" */
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import * as ActionsPhones from '../../actions/ActionsPhones'
import './style.scss';


class PhoneListItem extends React.PureComponent {
  render() {
    const { phone } = this.props;
    return (
      <button className="phone-list-item" onClick={() => this.props.selectPhone(phone)}>
        <img className="thumbnail" alt={phone.name} src={phone.image} />
        <p>{`${phone.brand} ${phone.name}`}</p>
      </button>
    );
  }
}

PhoneListItem.propTypes = {
  phone: PropTypes.object,
  selectPhone: PropTypes.func,
};


const matchDispatchToProps = (dispatch) => {
  return {
    selectPhone: (phone) => dispatch(ActionsPhones.selectPhone(phone)),
  }
}

export default connect(null, matchDispatchToProps)(PhoneListItem);
