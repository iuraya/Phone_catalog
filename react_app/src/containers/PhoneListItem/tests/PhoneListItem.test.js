/* eslint no-undef: "off" */
import React from 'react';
import { shallow } from 'enzyme';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import PhoneListItem from '../PhoneListItem';
import rootReducer from '../../../reducers';


const store = createStore(
  rootReducer,
  applyMiddleware(thunk),
);


describe('<PhoneListItem />', () => {
  const phone = {
    id: 1, name: 'test_phone_1', image: '', brand: 'test_brand_1', description: 'test_description_1', camera_description: 'test_camera_description_1',
  }

  it('render a phone Item', () => {
    const renderedComponent = shallow(<PhoneListItem phone={phone} selectPhone={() => {}} store={store} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
