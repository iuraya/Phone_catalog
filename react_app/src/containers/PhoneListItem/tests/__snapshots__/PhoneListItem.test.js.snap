// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhoneListItem /> render a phone Item 1`] = `
<PhoneListItem
  phone={
    Object {
      "brand": "test_brand_1",
      "camera_description": "test_camera_description_1",
      "description": "test_description_1",
      "id": 1,
      "image": "",
      "name": "test_phone_1",
    }
  }
  selectPhone={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
/>
`;
